m0t1fs/linux_kernel/dir.c

```diff
@@ -396,7 +396,11 @@ int m0t1fs_setxattr(struct dentry *dentry, const char *name,
 		/* Find optimal lid and set it to the inode.*/
 		layout_id = m0_layout_find_by_buffsize(&csb->csb_reqh.rh_ldom,
 							&ci->ci_pver, buffsize);
-		rc = m0t1fs_inode_set_layout_id(ci, &mo, layout_id);
+		if (layout_id > 0)
```

Comment:
can be simpler: 

rc = layout_id > 0
	? m0t1fs_inode_set_layout_id(ci, &mo, layout_id) 
	: M0_ERR_INFO(layout_id, "Could not find layout_id: "
		      "rc=%d", layout_id);

Reply:
It's rather the matter of taste. I prefer not to abuse the ternary operator.